### ISSUE TITLE
Xcode 10 + Swift 4.2 - UIAPI must be called from main thread

### ISSUE THREAD

**Issue by AAAstorga** [uias/Pageboy#176]
Hello,

I am getting this weird issue when using Pageboy.

It is in UIView+Localization.swift. I am not sure why this is a bug though, because you aren't updating any UI. 

Does anyone else have this issue?

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/15990084/46247498-3b4d2b00-c3da-11e8-84a8-d2026d7b1be7.png">

**Comment by msaps** [comment]
@AAAstorga never seen it before but will check it out!

**Comment by RamblinWreck77** [comment]
@msaps for problems like this I like to do something like the following:
```
func layoutIsRightToLeft() -> Bool {
   if Thread.isMainThread {
       return _layoutIsRightToLeft()
   } else {
   var result: Bool!
   DispatchQueue.main.sync { // Sync is safe because we are 100% not on the main thread
         result = _layoutIsRightToLeft()
   }
   return result
}

/// MAIN THREAD ONLY!
private func _layoutIsRightToLeft() -> Bool {
   // Do Main thread only work
}
```

This gives you the best performance since if you're on main you just go ahead and do the work, and only get the sync hit if you're off main thread.

**Comment by msaps** [comment]
@RamblinWreck77 @AAAstorga should be fixed in [3.0.1](https://github.com/uias/Pageboy/releases/tag/3.0.1) 🎉 

@RamblinWreck77 thanks for the tip!
